src/lib/sveltekit-search-params.svelte.ts

```diff
@@ -0,0 +1,363 @@
+import { browser, building } from '$app/environment';
+import { goto } from '$app/navigation';
+import { page as page_store } from '$app/stores';
+import type { Page } from '@sveltejs/kit';
+import { fromStore, readable, type Readable } from 'svelte/store';
+import type { EncodeAndDecodeOptions, StoreOptions } from './types';
+export type { EncodeAndDecodeOptions, StoreOptions };
+
+// during building we fake the page store with an URL with no search params
+// as it should be during prerendering. This allow the application to still build
+// and the client side behavior is still persisted after the build
+let page: Readable<Pick<Page, 'url'>>;
+if (building) {
+	page = readable({
+		url: new URL(
+			'https://github.com/paoloricciuti/sveltekit-search-params',
```

Comment:
assuming this url is never used for its origin the value might be ok, but a safer pick would be http://example.com

Reply:
Yeah the whole thing is actually never used at all i think...is just to prevent errors during building...but let me change that 😄
```suggestion
			'http://example.com',
```